server/coordinator/procedure/scatter.go

```diff
@@ -0,0 +1,178 @@
+	}
+	ScatterCallbacks = fsm.Callbacks{
+		EventScatterPrepare: func(event *fsm.Event) {
+			// FIXME: The following logic is used for static topology, which is a bit simple and violent.
+			// It needs to be modified when supporting dynamic topology.
+			request := event.Args[0].(*ScatterCallbackRequest)
+			c := request.c
+			handler := request.handler
+			nodeInfo := request.nodeInfo
+			ctx := request.cxt
+
+			if c.GetClusterState() == clusterpb.ClusterTopology_STABLE {
+				shardIDs, err := c.GetShardIDs(nodeInfo.GetEndpoint())
+				if err != nil {
+					event.Cancel(errors.WithMessage(err, "procedure scatterShard"))
+				}
+				if len(nodeInfo.GetShardInfos()) == 0 {
+					if err := handler.Dispatch(ctx, nodeInfo.GetEndpoint(), &schedule.OpenEvent{ShardIDs: shardIDs}); err != nil {
+						event.Cancel(errors.WithMessage(err, "procedure scatterShard"))
+					}
+				}
+			}
+
+			if !(int(c.GetClusterMinNodeCount()) <= c.GetNodesSize() &&
+				c.GetClusterState() == clusterpb.ClusterTopology_EMPTY) {
+				event.Cancel()
+			}
+
+			shardTotal := int(c.GetClusterShardTotal())
+			minNodeCount := int(c.GetClusterMinNodeCount())
+			perNodeShardCount := shardTotal / minNodeCount
+			shards := make([]*clusterpb.Shard, 0, shardTotal)
+			nodeList := make([]*clusterpb.Node, 0, c.GetNodesSize())
+			nodeCahce := c.GetClusterNodeCache()
+			for _, v := range nodeCahce {
+				nodeList = append(nodeList, v.GetMeta())
+			}
+
+			for i := 0; i < minNodeCount; i++ {
+				for j := 0; j < perNodeShardCount; j++ {
+					if i*perNodeShardCount+j < shardTotal {
+						// TODO: consider nodesCache state
+						shards = append(shards, &clusterpb.Shard{
+							Id:        uint32(i*perNodeShardCount + j),
+							ShardRole: clusterpb.ShardRole_LEADER,
+							Node:      nodeList[i].GetName(),
+						})
+					}
+				}
+			}
+
+			if err := c.UpdateClusterTopology(ctx, clusterpb.ClusterTopology_STABLE, shards); err != nil {
+				event.Cancel(errors.WithMessage(err, "procedure scatterShard"))
+			}
+
+			if err := c.Load(ctx); err != nil {
+				event.Cancel(errors.WithMessage(err, "procedure scatterShard"))
+			}
+
+			for nodeName, node := range nodeCahce {
+				if err := handler.Dispatch(ctx, nodeName, &schedule.OpenEvent{ShardIDs: node.GetShardIDs()}); err != nil {
+					event.Cancel(errors.WithMessage(err, "procedure scatterShard"))
+				}
+			}
+		},
+		EventScatterFailed: func(event *fsm.Event) {
+			// nolint
+		},
+		EventScatterSuccess: func(event *fsm.Event) {
+			// nolint
+		},
+	}
+)
+
+type ScatterProcedure struct {
+	id    uint64
+	state State
+
+	fsm      *fsm.FSM
+	c        *cluster.Cluster
+	nodeInfo *metaservicepb.NodeInfo
+	handler  *schedule.EventHandler
+}
+
+func NewScatterProcedure(cluster *cluster.Cluster, nodeInfo *metaservicepb.NodeInfo) *ScatterProcedure {
+	ScatterOperationFsm := fsm.NewFSM(
+		StateScatterBegin,
+		ScatterEvents,
+		ScatterCallbacks,
+	)
+
+	// TODO: fix id alloc
```

Comment:
Keep any comment as a statement as possible as it can. Here, we can put it like this:
```suggestion
	// TODO: try to allocate the procedure id in a proper way.
```